### PR TITLE
Add migration file specification

### PR DESCRIPTION
**Pull request by segalaj**
in addition of #357

### CODE REVIEW

File: documentation/en/source/pages/migration/migrations-spec/overview.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+Migrations contain all the information used to create migrations. They are JSON or YAML files, passed as a parameter to the hammr command-line.
+
+A migration has the following part:
+
+* ``migration``: defines the migration.
+
+.. toctree::
+   :titlesonly:
+
+   migration
```
Comment (cinlloc) [inline documentation/en/source/pages/migration/migrations-spec/overview.rst]: Should not be the reference to builders part here? Or elsewhere? (I cannot find it) Maybe a migration-target file?

File: documentation/en/source/pages/machine-images/builders/overview.rst [hunk 2]
```diff
@@ -2,8 +2,8 @@
 
 .. _template-builders:
 
-Builders
-========
+Builders Specifications
+=======================
 
 Within a template, the ``builders`` section is an array of objects, describing the list of machine images to build (and where possible publish). For example if you wished to build an AMI image for Amazon EC2 and another for Microsoft Azure, you would specify a builder for each.
```
[the remark this answers is not in the document]
Comment (emuus) [reply documentation/en/source/pages/machine-images/builders/overview.rst]: agree

File: documentation/en/source/pages/migration/migrations-spec/overview.rst (hunk 1 above)
Comment (emuus) [reply documentation/en/source/pages/migration/migrations-spec/overview.rst]: agree

File: documentation/en/source/pages/migration/migrations-spec/overview.rst [hunk 3]
```diff
@@ -0,0 +1,17 @@
+.. Copyright (c) 2007-2018 UShareSoft, All rights reserved
+
+.. _migrations-spec:
```
[the remark this answers is not in the document]
Comment (emuus) [reply documentation/en/source/pages/migration/migrations-spec/overview.rst]: agree

File: documentation/en/source/pages/machine-images/builders/overview.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (emuus) [reply documentation/en/source/pages/machine-images/builders/overview.rst]: Do we need to add a note to indicate which targets are supported for Migration ?

File: documentation/en/source/pages/templates/templates-spec/overview.rst [hunk 4]
```diff
@@ -16,4 +16,4 @@ A template has two main parts:
    :titlesonly:
 
    stack
-   builders/overview
\ No newline at end of file
+   ../../machine-images/builders/overview
```
Comment (emuus) [inline documentation/en/source/pages/templates/templates-spec/overview.rst]: does this build correctly? I think the reference should be in the intro:
builders: an array defining the format of the machine images to build. For more information on builders, refer to :ref:`template-builders`
Same info should be added to the new section "Migration Specifications"

**Comment by segalaj** [comment]
will be covered by #357

**Pull request closed without merging**